src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -515,13 +515,19 @@ std::vector<SMALL_RECT> Terminal::_GetSelectionRects() const
     }
 
     // Add anchor offset here to update properly on new buffer output
-    SHORT temp1, temp2;
-    THROW_IF_FAILED(ShortAdd(_selectionAnchor.Y, _selectionAnchor_YOffset, &temp1));
-    THROW_IF_FAILED(ShortAdd(_endSelectionPosition.Y, _endSelectionPosition_YOffset, &temp2));
+    SHORT y1;
+    SHORT y2;
+    THROW_IF_FAILED(ShortAdd(_selectionAnchor.Y, _selectionAnchor_YOffset, &y1));
+    THROW_IF_FAILED(ShortAdd(_endSelectionPosition.Y, _endSelectionPosition_YOffset, &y2));
+
+    // clamp X values to be within buffer bounds
+    const auto bufferWidth = _buffer->GetSize().RightInclusive();
+    SHORT x1 = std::clamp(_selectionAnchor.X, static_cast<SHORT>(0), bufferWidth);
+    SHORT x2 = std::clamp(_endSelectionPosition.X, static_cast<SHORT>(0), bufferWidth);

```

Comment:
So out of curiosity, if you maximize, then make a selection, then restore down, then re-maximize, the entire selection remains selected, correct?

Is this something we want?

Reply:
1. **Yes**: The clamping happens at render time. So, the same selection would be in effect when you re-maximize it (assuming it's the same dimensions as before).

2. **Yes?**: We definitely want this PR because it fixes a crash that's really easy to get. (if you make a selection of the entire screen, it crashes w/out this change)

Where it gets weird is with wrapping text. Then, I think you won't really know what you've selected because the (x,y) position of the anchors stay the same, but the text changed locations. 

With regards to THAT issue I just mentioned, we have a few options I can think of:
(1) clear selection on resize - I think this is the easy way out but, I guess it works?
(2) update selection anchors on resize - we'd have to identify when text is (not) wrapping and update appropriately

Reply:
We could theoretically reflow the selection anchors the same way we attempt to reflow the cursor position when we handle a resize-with-reflow in conhost. (This is not code we have in Terminal today). I will add, that code is a bug factory, always has been, and I believe it always will be.

Though, conhost also just dismisses the selection so it doesn't have to deal with this problem. That's almost certainly not something would complain about, _right_?